lib/adapter.js

```diff
@@ -0,0 +1,494 @@
+	amapi.on('heap', function(res) {
+		heapMetrics.total = +res.total;
+		heapMetrics.used = +res.used;
+		heapMetrics.count++;
+	});
+}
+
+function createGCListener() {
+	amapi.on('gc', function(res) {
+		gcMetrics.used = +res.used;
+		gcMetrics.count++;
+	});
+}
+
+function createHTTPListener() {
+	amapi.on('http', function(res) {
+		// All we need to do is increment the count
+		httpMetrics.count++;
+	});
+}
+
+function createMessageListener() {
+	// Map the event types from axon and strong mq to whether they 
+	// were send or receive events.
+	var incrementCounts = function (data) {
+		var types = {push : 'send', pull: 'receive',
+				pub : 'send', sub : 'receive',
+				req : 'send', rep : 'receive',
+				'pub-emitter' : 'send', 'sub-emitter' : 'receive'};
+		if( types[data.type] == 'send' ) {
+            if (messageMetrics.sent != undefined) {
+                messageMetrics.sent++;   
+            } else {
+                messageMetrics.sent = 1;
+            }
+			
+		} else if( types[data.type] == 'receive' ) {
+			if (messageMetrics.received != undefined) {
+                messageMetrics.received++;
+            } else {
+                messageMetrics.received = 1;
+            }
+		}
+	}
+	amapi.on('strong-mq', incrementCounts);
+	amapi.on('axon', incrementCounts);
+}
+
+function createObjectListener() {
+	amapi.on('objectType', function(objectType) {
+
+	});
+	amapi.on('objectSize', function (objectSize) {
+		
+	});
+}
+
+function createProbeListener(probeName){
+	amapi.on(probeName, function(res){
+		var duration = res.duration;
+		var currentMetrics;
+		
+		// If no metrics object exists, create one
+		if (probeMetrics[probeName] == undefined){
+			probeMetrics[probeName] = {count: 0, runningTotal: 0};
+		}
+		
+		// Update all metrics accordingly
+		updateMetrics(probeName, duration); 
+	});
+}
+
+function getEventLoopData() {
+	// Calculate average
+	var count = eventLoopMetrics.count;
+	var average = eventLoopMetrics.averageRunningTotal / count;
+
+	// Build JSON
+	var eventLoopEvent = {
+		"count" : count,
+		"minimum" : eventLoopMetrics.min,
+		"maximum" : eventLoopMetrics.max,
+		"average" : average
+	};
+
+	// Reset values
+	eventLoopMetrics.count = 0;
+	eventLoopMetrics.min = null;
+	eventLoopMetrics.max = null;
+	eventLoopMetrics.averageRunningTotal = 0;
+
+	return eventLoopEvent
+}
+
+function getCpuData() {
+	var system = cpuMetrics.system / cpuMetrics.count;
+	var user = cpuMetrics.user / cpuMetrics.count;
+	var cpuEvent = {
+		"system" : system,
+		"user" : user,
+		"total" : system + user
+	};
+
+	// Reset values
+	cpuMetrics.system = 0;
+	cpuMetrics.user = 0;
+	cpuMetrics.count = 0;
+
+	return cpuEvent;
+}
+
+function getHTTPData() {
+	var httpEvent = {
+		"count" : httpMetrics.count
+	}
+
+	// Reset Count
+	httpMetrics.count = 0;
+
+    return httpEvent
+}
+
+function getProbeData(probeName){
+	
+	//Get data for given probe
+	var probeData = probeMetrics[probeName];
+	var count = probeData.count;
+	var average;
```

Comment:
`var average = count ? probeData.runningTotal / count : 0` would compress 5 lines into 1 (we aren't members of the "ternary operators are evil" cult).